Wikipedia/UI-V5/WMFNearbyViewModel.h

```diff
@@ -0,0 +1,130 @@
+//
+//  WMFNearbyViewModel.h
+//  Wikipedia
+//
+//  Created by Brian Gerstle on 9/9/15.
+//  Copyright (c) 2015 Wikimedia Foundation. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@class MWKTitle;
+@class WMFLocationManager;
+@class WMFLocationSearchFetcher;
+@class WMFLocationSearchResults;
+@class WMFNearbyViewModel;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ *  Object which displays location search results and errors.
+ */
+@protocol WMFNearbyViewModelDelegate <NSObject>
+
+/**
+ *  Invoked when a view model has fetched new search results for a location.
+ *
+ *  This can happen as a result of either:
+ *
+ *    - Site updates (e.g. user changes the Wiki to get results from)<br/>
+ *    - The user's location changing a significant amount
+ *
+ *  @param viewModel The view model which was updated.
+ *  @param results   The results of the fetch.
+ */
+- (void)nearbyViewModel:(WMFNearbyViewModel*)viewModel didUpdateResults:(WMFLocationSearchResults*)results;
+
+- (void)nearbyViewModel:(WMFNearbyViewModel*)viewModel didFailWithError:(NSError*)error;
+
+// TODO: separate callback for authorization prompts
+
+@end
+
+@interface WMFNearbyViewModel : NSObject
```

Comment:
I'm not sure about calling this a view model - is it one? 

One responsibility of a view model should be to translate model data into a form that can be displayed on screen. But that is done elsewhere - mostly in the cell classes: that is where strings are created and formatted and where bearings and locations are turned into numbers.

Another way to put it is that view models should be able to act as a proxy for testing the UI… "given this model object, what is the string to be displayed on screen for the title label." At tis point only the cell could actually answer that question.

This seems more like a controller that is coordinating the fetching of models and some location logic to yield models (Search Results). 

Moving to MVVM may be a good idea… but we should be careful in how we name things as we transition to that type of architecture. Calling this a view model sets up expectations that this class doesn't quite hold up to. For now it makes more sense to keep with non-MVVM terminology until we can actually implement something that is a true view model.